### PR TITLE
Refs [TASK][36907] Unit test detail #24

### PR DESCRIPTION
**Pull request by hhd0307**
## Related Tickets
-  [#Task 36907: Unit test detail](https://edu-redmine.sun-asterisk.vn/issues/36907)
## WHAT
- create unit test
## Evidence (Screenshot or Video)
Screenshot from 2021-05-31 10-39-31

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

**Review by trinq-2471 — APPROVED** [review]

**Review by khoaqh-2470 — APPROVED** [review]

File: app/src/test/java/com/sun/hero_01/ui/detail/DetailPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    @Mock
+    private lateinit var exception: Exception
+
+    private lateinit var presenter: DetailPresenter
+    private val detail = FakeData.HERO_DETAIL
+
+    @Before
+    fun setUp() {
+        presenter = DetailPresenter(repository, favoriteRepository)
+        presenter.setView(view)
```
Comment (tuanlvt-1117) [inline app/src/test/java/com/sun/hero_01/ui/detail/DetailPresenterTest.kt]: scope function

File: app/src/main/java/com/sun/hero_01/utils/FakeData.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package com.sun.hero_01.utils
+
+import com.sun.hero_01.data.model.HeroDetail
+
+object FakeData {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun/hero_01/utils/FakeData.kt]: data test e có thể đưa vào trong pakage test cho đồng bộ nhé

File: app/src/test/java/com/sun/hero_01/ui/detail/DetailPresenterTest.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Mock
+    private lateinit var exception: Exception
+
+    private lateinit var presenter: DetailPresenter
+    private val detail = FakeData.HERO_DETAIL
+
+    @Before
+    fun setUp() {
+        presenter = DetailPresenter(repository, favoriteRepository).apply { setView(view) }
```
Comment (tuanlvt-1117) [inline app/src/test/java/com/sun/hero_01/ui/detail/DetailPresenterTest.kt:39]:
```suggestion
        presenter = DetailPresenter(repository, favoriteRepository).run { setView(view) }
```

Comment (hhd0307) [reply app/src/test/java/com/sun/hero_01/ui/detail/DetailPresenterTest.kt:39]: Dùng run sẽ có kiểu trả về là kiểu của hàm setView(), nó trả về Unit 
Như vậy không thể set cho presenter kiểu DetailPresenter được, nên e mới xài apply ạ.

**Pull request merged**